### DISCUSSION TITLE
Update to 11.1.0 - missing Sonoff Zbridge (75)

### DISCUSSION THREAD

**Discussion by huehnerhose** [arendst/Tasmota#15521] [Q&A] [answered]
Hi, 

I had a working setup with a Sonoff Zigbee Bridge running on tasmota 11.0.0. 
Because newer is always better I clicked on firmware update - OTA - do it. Without any second thought. 
After the restart my paired sensors weren't visible but only a "Toggle On/off" thing. 

Since I am not blessed with deep understanding of the inner workings of Tasmota I checked the install guide and it seems the module type Sonoff Zbbrdige (75) is missing and I should select that one. 

Any thoughts how I can get it back?

Thanks a lot!

**Comment by barbudor** [comment, marked as answer]
On the information page me you can check which variant of Tasmota has been installed
I would guess that your OtaUrl was not pointing to tasmota-zbbridge.bin.gz but to the default tasmota.bin.gz

Go to the upgrade page, be sure to have the right url there and press again update
The url should be http://ota.tasmota.com/tasmota/release/tasmota-zbbridge.bin.gz

**Reply by barbudor** [reply]
Note that you may have to set back manually module 75

**Reply by huehnerhose** [reply]
Many thanks. Solved my problem - didn't even had to manually select module 75 👍

**Comment by oywino** [comment]
Can you explain what the command "Module 75" actually does, and why/when it is necessary and when it isn't?

**Reply by sfromis** [reply]
I'd suggest you to not at all use the command `Module 75` as that is implied when you use the right template from https://templates.blakadder.com/sonoff_ZBBridge.html
Doing that also makes sure that you have the gpio pins configured as appropriate. Note that it is not the same if you have the newer "pro" version of the Sonoff Zigbee bridge. In the template, you see the module number as `"BASE":75`.

What module 75 does (also when implied by the template) is to enable runtime support for Zigbee2Tasmota. Hence it is important to have the right template configured, and a different template (without module 75) if you instead want to use ZHA mode.

**Comment by oywino** [comment]
I did the `Module 75` because Lewis (Everything SmartHome)  recommended it in one of his tutorial videos. But I guess it's no harm, just superfluous, right?  No need to try to "undo" things?
I'd like to use my Sonoff Tasmotized ZigBee Bridge with Z2M (and not ZHA). Which template is the right one then?

**Reply by sfromis** [reply]
ZHA mode is just a ser2net tunnel to the Zigbee chip, seen from Tasmota Z2M is just the same. And then you need the template without module 75, as described on the page I linked.

**Comment by oywino** [comment]
Ok, so I've installed Z2M but it fails to start something called "zigbee-herdsman" according to the log:

```
debug 2023-10-31 22:03:55: Can't load state from file /app/data/state.json (doesn't exist)
info  2023-10-31 22:03:55: Logging to console and directory: '/app/data/log/2023-10-31.22-03-55' filename: log.txt
info  2023-10-31 22:03:55: Starting Zigbee2MQTT version 1.33.1 (commit #7e63039)
info  2023-10-31 22:03:55: Starting zigbee-herdsman (0.19.0)
debug 2023-10-31 22:03:55: Using zigbee-herdsman with settings: '{"adapter":{"concurrent":null,"delay":null,"disableLED":false},"backupPath":"/app/data/coordinator_backup.json","databaseBackupPath":"/app/data/database.db.backup","databasePath":"/app/data/database.db","network":{"channelList":[11],"extendedPanID":[221,221,221,221,221,221,221,221],"networkKey":"HIDDEN","panID":6754},"serialPort":{"path":"tcp://192.168.117:8888"}}'
error 2023-10-31 22:06:06: Error while starting zigbee-herdsman
error 2023-10-31 22:06:06: Failed to start zigbee
error 2023-10-31 22:06:06: Check https://www.zigbee2mqtt.io/guide/installation/20_zigbee2mqtt-fails-to-start.html for possible solutions
error 2023-10-31 22:06:06: Exiting...
error 2023-10-31 22:06:06: Error: Error while opening socket
    at Socket.<anonymous> (/app/node_modules/zigbee-herdsman/src/adapter/z-stack/znp/znp.ts:198:24)
    at Socket.emit (node:events:526:35)
    at emitErrorNT (node:internal/streams/destroy:151:8)
    at emitErrorCloseNT (node:internal/streams/destroy:116:3)
    at processTicksAndRejections (node:internal/process/task_queues:82:21)

```
I've installed Z2M on a QNAP NAS in a docker container. Here's my compose yaml:

```
version: '3'
services:
  zigbee2mqtt:
    image: koenkk/zigbee2mqtt:latest
    restart: "no"
    ports:
      - 9080:8080
    environment:
      TZ: Europe/Oslo
    volumes:
      - /share/Public/Z2M:/app/data
      - /share/Public/Z2M/run/udev:/run/udev:ro
 
```
My ZigBee device is the Sonoff ZbBridge flashed with Tasmota 13.2.0 and the ncp-uart-nsw_6.7.9_115200 firmware. I'm using Blackadders ZHA Template and the Bridge runs just fine and the WebUI console shows all activities. Here's my Z2M _configuation.yaml_ file:

```
homeassistant: true
mqtt:
  base_topic: zigbee2mqtt
  server: mqtt://192.168.1.8:1883
serial:
  port: tcp://192.168.117:8888
  adapter: ezsp
frontend: true
advanced:
  log_level: debug
  network_key: GENERATE
```

My MQTT server is Mosquitto 2.0.4 and it has been running for a long time and works well.
Any help to resolve my issue is much apreciated.

**Reply by sfromis** [reply]
Looks like your're using port 8888 without allowing it through to your container. And your `port: tcp://192.168.117:8888` is badly wrong, cannot be a valid IP address.

**Comment by oywino** [comment]
Oh - I'm sure you're on to something, but I do not understand what you mean by "badly wrong" ?  Is it a syntax error?
The address 192.168.1.117 is the address to the Sonoff ZigBee Bridge.

I just copied the instructions from this guide:
![image](https://github.com/arendst/Tasmota/assets/18447119/eb6c2888-819e-4b82-8b44-8928f046239a)
.. which I found here: https://thesmarthomejourney.com/2022/07/19/zigbee2mqtt-quick-start-guide/#Installing_Home_Assistant_MQTT_and_Zigbee2MQTT

**Reply by barbudor** [reply]
But you didn't typed it correctly in your yaml

**Comment by oywino** [comment]
Didn't I ?!?  I must be blind. Please tell me exactly where my typo is, please - as I cannot for the life of me see any misspellings or wrong indentations or anything. I followed the guide as exactly as I could here: https://www.zigbee2mqtt.io/guide/configuration/adapter-settings.html

**Reply by sfromis** [reply]
You typed `192.168.117` instead of `192.168.1.117`. That's badly wrong, cannot be a valid IP address. :stuck_out_tongue_closed_eyes:

**Comment by oywino** [comment]
OMG 👀  You are an angel!!!   How could I be so blind?  Thank you so very much 💕
